### PR TITLE
Feature/modular filehandling

### PR DESCRIPTION
**Pull request by klaseskilson**
Move file adding and downloading logic to the `Identity` model. The idea behind this is that the difference there might be between the different services, such as Dropbox, our own local storage and Drive _will be in which Identity the file belongs to_.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -35,6 +37,8 @@ gem 'font-awesome-rails'
 gem 'font-awesome-sass'
 # sass mixins that makes you happy
 gem 'bourbon'
+# normalize!
+gem 'normalize-rails'
```
Comment (klaseskilson) [inline Gemfile:41]: Normalize resets the css, so that any Browser-specific special default css styles is removed.

**Comment by tkomstadius** [comment]
You can only add folders, not remove them

**Comment by klaseskilson** [comment]
Yes, that is correct. Working on it!

**Comment by ronjagrosz** [comment]
Adding a folder works, but the minus sign does not appear for the individual files in the folder. Pressing the plus sign gives the feedback that the file already exists. :(

**Comment by klaseskilson** [comment]
@tkomstadius, do you have the same issue as @Fluffpannkaka?

**Comment by tkomstadius** [comment]
No, I don't think so. I still haven't got a minus for the folder but if i add then collapse all the individual files have a minus. But when i remove files, add the folder again the removed files isn't added again.

**Comment by ronjagrosz** [comment]
It works now! :D only seems to be a refresh issue. When adding the folder, expanding, the minus signs are there. If I then click the minus button on the folder the feedback in the console says that they are removed, but the signs do not return to plus on the individual files, they still have(-). Only for the folder, which now has (+). Maybe do some angular for refreshing the icons(?)

**Comment by tkomstadius** [comment]
I can't remove files anymore. They just gets added multiple times instead

**Comment by tkomstadius** [comment]
I also get a lot of "could not unsave file" (404) when I try to remove them.

**Comment by ronjagrosz** [comment]
Ah no. I can't add files either.

**Comment by klaseskilson** [comment]
Found a couple of more issues, too. When removing files, they still exist in `$scope.bruse_files`. When removing folders, the files in that folder still exist in `$scope.bruse_file` _and_ on the file remote file object in `$scope.files`.

The icons totally depend on the remote file object.

**Comment by klaseskilson** [comment]
This could possibly be resolved using the `collect` method in `FileHandler`, since that's an easy way to make sure we have all existing files loaded and then the [`findFile` method in `FileCtrl`](https://github.com/tkomstadius/bruse/blob/5ca4716301cd43a0b6d1aacf15c09d844dc296d9/app/assets/javascripts/controllers/FileCtrl.coffee#L84). 

Much like we [do when the page loads](https://github.com/tkomstadius/bruse/blob/5ca4716301cd43a0b6d1aacf15c09d844dc296d9/app/assets/javascripts/controllers/FileCtrl.coffee#L9-L22) and when we [expand a folder](https://github.com/tkomstadius/bruse/blob/5ca4716301cd43a0b6d1aacf15c09d844dc296d9/app/assets/javascripts/controllers/FileCtrl.coffee#L38-L39).

**Comment by klaseskilson** [comment]
I feel this part of the system is consuming too much development time. Most of the features in this PR is about importing external files into Bruse, and not handling the actual BruseFiles. Therefore, I think we can merge this PR into develop, and then branch out to implement batch (multiple) handling of the existing BruseFiles. I suggest that we also try to implement a way to separate the already existing files and the newly added files to show the user what's happening.

Does that sound OK?

**Comment by tkomstadius** [comment]
Yes, if that will help us move forward I think that that sounds like a good plan. Then I feel this is ready for merge.

**Comment by danielronnkvist** [comment]
I think it looks good!

**Pull request merged**